src/main/kotlin/de/cyface/collector/auth/MockedHandlerBuilder.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2023 Cyface GmbH
+ *
+ * This file is part of the Cyface Data Collector.
+ *
+ * The Cyface Data Collector is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Cyface Data Collector is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the Cyface Data Collector. If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.cyface.collector.auth
+
+import de.cyface.collector.handler.HTTPStatus
+import io.vertx.core.Future
+import io.vertx.core.json.JsonObject
+import io.vertx.ext.auth.impl.UserImpl
+import io.vertx.ext.web.Route
+import io.vertx.ext.web.RoutingContext
+import io.vertx.ext.web.handler.OAuth2AuthHandler
+import java.util.UUID
+
+/**
+ * Mocked OAuth2 builder which creates an OAuth2 handler for testing.
+ *
+ * @author Armin Schnabel
+ * @version 1.0.0
+ * @since 7.0.0
+ */
+class MockedHandlerBuilder : OAuth2HandlerBuilder {
+
+    override fun create(): Future<OAuth2AuthHandler> {
+        val handler: OAuth2AuthHandler = object : OAuth2AuthHandler {
+            override fun handle(event: RoutingContext) {
+                // FIXME: FileUploadTest ...withWrongCredentials expects this handler to return 401
+                // Testing the Mocked Auth Handler does not make much sense, so just remove the two tests (and this)?
+                val invalidCredentials = event.request().headers().get("Authorization").equals("Bearer invalidToken")
+                if (invalidCredentials) {
+                    event.fail(HTTPStatus.UNAUTHORIZED)
+                }
```

Comment:
This makes not much sense. The question is if we should just remove the two `FileUploadTests` which check if invalid credentials lead to an 401 error. We can't use a real Keycloak server in the test, so we use this Mocked handler here, but it this just returns 401 we don't need to test this really.

Reply:
Remove it. We are not going to test third party software.